contributors/design-proposals/storage/csi-migration.md

```diff
@@ -287,8 +287,31 @@ existing Pods in the ADC.
 
 
 ### Volume Resize
+#### Offline Resizing
+In the OperationGenerator, `GenerateExpandVolumeFunc` is used to expand a volume after 
+Pod is deleted. At the beginning of the `GenerateExpandVolumeFunc`, we will check whether 
+the volume supports migration and whether migration is enabled. If enabled, we will return
+a noop function to by-pass expand controller.
 
-TODO: Design
+In the OperationGenerator, `GenerateMountVolumeFunc` is used to expand file system after volume
+is expanded and staged/mounted. The migration logic is covered by previous migration of volume mount.
+
+For external resizer, we will update it to expand volume for both CSI volume and in-tree 
+volume (only if migration is enabled). For migrated in-tree volume, it will update in-tree PV object
+with new volume size and mark in-tree PVC as resizing finished.
+
+To synchronize between expand controller and external resize, external resizer will find provisioner name
+using PVC annotation `volume.beta.kubernetes.io/storage-provisioner`. If the name matches to the CSI driver
+name, it guarantees that the PV is provisioned by CSI driver, hence external resizer will proceed with
+volume resizing. Otherwise, it will yield to expand controller.
+
+//TODO: expand block volume
+
+Caveat: for static provisioning, user will need to manually add the `volume.beta.kubernetes.io/storage-provisioner`
```

Comment:
this is a change from existing functionality. Could we maybe annotate the PVC with an admission controller to make sure the user doesn't have to do any additional actions?

/cc @msau42

Reply:
If we don't want users to need to modify something, then perhaps it's better for the in-tree resize controller to add its own annotation (which I thought @gnufied was doing anyway)

Reply:
this would be a good solution. It can add an annotation then yield to the `external-resizer` when migration is on

Reply:
yeah I think we will need a new annotation for resizing. In general that is a good idea.

Reply:
Updated to use `volume.kubernetes.io/storage-resizer`